core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -304,6 +316,42 @@ class WorkspaceService(protected val userInfo: UserInfo,
     })
   }
 
+  private def getAzureCloudContextFromWorkspaceManager(workspaceContext: Workspace, parentSpan: Span = null) = {
+    workspaceContext.workspaceType match {
+      case WorkspaceType.McWorkspace => {
+        val span = startSpanWithParent("getWorkspaceFromWorkspaceManager", parentSpan)
+
+        try {
+          val wsmInfo = workspaceManagerDAO.getWorkspace(workspaceContext.workspaceIdAsUUID, userInfo.accessToken)
+
+          Option(wsmInfo.getAzureContext) match {
+            case Some(azureContext) => Some(WorkspaceAzureCloudContext(
+              azureContext.getTenantId,
+              azureContext.getSubscriptionId,
+              azureContext.getResourceGroupId)
+            )
+            case None => None
+          }
+          // todo pull in GCP context here
```

Comment:
I'm not clear what this TODO refers to. Can you provide some more details?

Reply:
A WSM workspace can have either a GCP cloud context or an Azure cloud context. For now, all this code path knows about is Azure cloud contexts, but in the long term (once we have a migration plan in place), a GCP context will be retrieved the same way.

That being said, this is explicitly a method to get azure data from WSM, so this comment is confusing, will remove.